Our.Umbraco.TagHelpers/IsActivePageTagHelper.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.AspNetCore.Razor.TagHelpers;
+using Umbraco.Cms.Core.Web;
+using Umbraco.Extensions;
+
+namespace Our.Umbraco.TagHelpers
+{
+    /// <summary>
+    /// An attribute TagHelper that only works with <a> tags
+    /// 
+    /// Applying the attribute our-is-active-page="navi-active"
+    /// Will apply the CSS class name navi-active to the class attribute
+    /// if the value in the href of the <a> compared to the page being rendered
+    /// is the current page or part of an ancestor
+    /// </summary>
+    [HtmlTargetElement("a", Attributes = "our-is-active-page")]
+    public class IsActivePageTagHelper : TagHelper
+    {
+        private IUmbracoContextAccessor _umbracoContextAccessor;
+
+        public IsActivePageTagHelper(IUmbracoContextAccessor umbracoContextAccessor)
+        {
+            _umbracoContextAccessor = umbracoContextAccessor;
+        }
+
+        /// <summary>
+        /// The CSS class name you wish to append to the class attribute
+        /// on an <a> tag when the page is active/selected
+        /// </summary>
+        [HtmlAttributeName("our-is-active-page")]
+        public string ActiveClassName { get; set; }
+        
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            // Remove the attribute 
+            // We don't want it in the markup we send down to the page
+            if(output.Attributes.TryGetAttribute("our-is-active-page", out TagHelperAttribute tagHelperAttr))
+            {
+                output.Attributes.Remove(tagHelperAttr);
+            }
+
+            var ctx = _umbracoContextAccessor.GetRequiredUmbracoContext();
+
+            // On the <a> try to find the href attribute and its value
+            var href = output.Attributes["href"]?.Value.ToString();
+            if (string.IsNullOrEmpty(href))
+            {
+                return;
+            }
+
+            // Clean href values of any querystrings
+            // As we won't be able to query GetByRoute for an Umbraco node with them
+            href = href.Substring(0, href.IndexOf("?") > 0 ? href.IndexOf("?") : href.Length);
```

Comment:
Yep you  are right, it blows up if the URLs are absolute
`<a href="@item.Url(mode: UrlMode.Absolute)" class="nav-link" our-active-class="nav-link--active">@item.Name</a>`

Any suggestions ?

Reply:
I think the best way would be to parse it to an `Uri` instance and get the path. This can only be retrieved from absolute URIs, so relative links should be turned into absolute URIs by using the current page as base (which also makes `href="contact"` resolve to the correct path):

```c#
if (Uri.TryCreate(href, UriKind.Absolute, out var link) ||
    Uri.TryCreate(ctx.PublishedRequest.Uri, href, out link))
{
    var linkContent = ctx.Content.GetByRoute(link.AbsolutePath);
    if (linkContent != null && linkContent.IsAncestorOrSelf(ctx.PublishedRequest.PublishedContent))
    {
        output.AddClass(ActiveClassName, HtmlEncoder.Default);
    }
}
```